packages/react-components/src/atoms/TextField.tsx

```diff
@@ -152,9 +160,7 @@ const TextField: React.FC<TextFieldProps> = ({
   customIndicatorPosition = 'right',
   loading = false,
   indicateValid = customIndicator === undefined &&
-    (required !== undefined ||
-      maxLength !== undefined ||
-      pattern !== undefined),
+    (pattern !== undefined || showValidationType.has(type)),
```

Comment:
The `has` check here needs a test, one example each i.e. `email` shows indicator even without `pattern` or `indicateValid` set, `text` does not show indicator without `pattern` or `indicateValid` set